src/PulseWelder/src/speaker.cpp

```diff
@@ -140,9 +140,10 @@ void Speaker::addDigitSounds(uint32_t val) {
             val /= 10; // shift to next digit
           }
           if (count == 0) { count = 1; digits[0] = 0; }
-          // special case for value == 0 
-
-          for (; count; count-- ) { AddNumberToSequence(digits[count]); }
+          // special case for value == 0
+          for (; count; count-- ) {
+            AddNumberToSequence(digits[count-1]);
```

Comment:
Once you mentioned an issue with the number announcement, I immediately thought I did this off-by-one mistake. Well spotted!

Reply:
We've all done that.